src/Renci.SshNet/Security/Cryptography/Ciphers/AesGcmCipher.cs

```diff
@@ -0,0 +1,105 @@
+﻿#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_0_OR_GREATER
+using System;
+using System.Security.Cryptography;
+
+using Renci.SshNet.Common;
+
+namespace Renci.SshNet.Security.Cryptography.Ciphers
+{
+    /// <summary>
+    /// AES GCM cipher implementation.
+    /// <see href="https://datatracker.ietf.org/doc/html/rfc5647"/>.
+    /// </summary>
+    public sealed class AesGcmCipher : AeadCipher, IDisposable
+    {
+        private readonly byte[] _nonce;
+        private readonly AesGcm _aesGcm;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AesGcmCipher"/> class.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <param name="iv">The IV.</param>
+        public AesGcmCipher(byte[] key, byte[] iv)
+            : base(key, tagSize: 16)
+        {
+            _nonce = iv.Take(12);
+#if NET8_0_OR_GREATER
+            _aesGcm = new AesGcm(key, TagSize);
+#else
+            _aesGcm = new AesGcm(key);
+#endif
+        }
+
+        /// <inheritdoc/>
+        public override byte[] Encrypt(byte[] input, int offset, int length)
+        {
+            // [outbound sequence][packet length field][padding length field sz][payload][random paddings]
+            // [--4 bytes(offset)][------4 bytes------][-------------------Plain Text--------------------]
+            var packetLengthField = new ReadOnlySpan<byte>(input, offset, 4);
+            var plainText = new ReadOnlySpan<byte>(input, offset + 4, length - 4);
+
+            var cipherText = new byte[length - 4];
+            var tag = new byte[TagSize];
+
+            _aesGcm.Encrypt(_nonce, plainText, cipherText, tag, packetLengthField);
```

Comment:
You can avoid the BlockCopy() calls below by using a ReadOnlySpan for the cipherText and tag ouputs as well. This considerably increases performance, while also allocating only one buffer instead of two.